File: 2016/finale/enciclopedia/att/enciclopedia.c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

#define MAXN 10000
#define MAXC 30
FILE *fr, *fw;

void volume(char* S) {
    fprintf(fw, "%s\n", S);
}

void rilega(int N, int K, char* parole[]) {
    // Mettete qui il codice della soluzione
    char a[] = "QUA-RAN";
    char b[] = "TA-DUE";
    volume(a);
    volume(b);
}

char* parole[MAXN];

int main() {
    int N, K, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &K));
    for(i=0; i<N; i++) {
        parole[i] = (char*) malloc(sizeof(char) * (MAXC + 1));
        assert(0 < fscanf(fr, "%s", parole[i]));
    }

    rilega(N, K, parole);

    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/finale/enciclopedia/att/enciclopedia.pas
var
  fr, fw: text;

procedure volume(var S: ansistring);
begin
    writeln(fw, S);
end;

procedure rilega(N, K: longint; var parole: array of ansistring);
var a, b : ansistring;
begin
    (* Mettete qui il codice della soluzione *)
    a := 'QUA-RAN';
    b := 'TA-DUE';
    volume(a);
    volume(b);
end;


const
    MAXN = 10000;
var
    N, K, i : longint;
    parole  : array[0..MAXN-1] of ansistring;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, K);
    for i:=0 to N-1 do
        read(fr, parole[i]);

    rilega(N, K, parole);

    close(fr);
    close(fw);
end.
